packages/qwik-city/runtime/src/router-outlet-component.ts

```diff
@@ -2,59 +2,43 @@ import {
   component$,
   jsx,
   type JSXNode,
-  SkipRender,
   useContext,
-  useOnDocument,
   _IMMUTABLE,
   _jsxBranch,
-  event$,
+  type PropFunction,
+  jsxs,
+  Fragment,
 } from '@builder.io/qwik';
-import type { ClientHistoryWindow } from './client-navigate';
 import { ContentInternalContext } from './contexts';
+import type { RestoreScroll } from './types';
+import { SinglePageNavigation } from './single-page-navigation-component';
+import { toLastPositionOnPopState } from './scroll-restoration';
 
 /**
  * @public
  */
-export const RouterOutlet = component$(() => {
-  _jsxBranch();
+export interface RouterOutletProps {
+  restoreScroll$?: PropFunction<RestoreScroll>;
+}
 
-  useOnDocument(
-    'qinit',
-    event$(() => {
-      const POPSTATE_FALLBACK_INITIALIZED = '_qCityPopstateFallback';
-      const CLIENT_HISTORY_INITIALIZED = '_qCityHistory';
+/**
+ * @public
+ */
+export const RouterOutlet = component$<RouterOutletProps>(
+  ({ restoreScroll$ = toLastPositionOnPopState }) => {
+    _jsxBranch();
 
-      if (!(window as ClientHistoryWindow)[POPSTATE_FALLBACK_INITIALIZED]) {
-        (window as ClientHistoryWindow)[POPSTATE_FALLBACK_INITIALIZED] = () => {
-          if (!(window as ClientHistoryWindow)[CLIENT_HISTORY_INITIALIZED]) {
-            // possible for page reload then hit back button to
-            // navigate to a client route added with history.pushState()
-            // in this scenario we need to reload the page
-            location.reload();
-          }
-        };
+    const { value } = useContext(ContentInternalContext);
+    let contentCmp: JSXNode | null = null;
 
-        setTimeout(() => {
-          // this popstate listener will be removed when the client history is initialized
-          addEventListener(
-            'popstate',
-            (window as ClientHistoryWindow)[POPSTATE_FALLBACK_INITIALIZED]!
-          );
-        }, 0);
+    if (value && value.length > 0) {
+      for (let i = value.length - 1; i >= 0; i--) {
+        contentCmp = jsx(value[i].default, { children: contentCmp });
       }
-    })
-  );
-
-  const { value } = useContext(ContentInternalContext);
-  if (value && value.length > 0) {
-    const contentsLen = value.length;
-    let cmp: JSXNode | null = null;
-    for (let i = contentsLen - 1; i >= 0; i--) {
-      cmp = jsx(value[i].default, {
-        children: cmp,
-      });
     }
-    return cmp;
+
+    return jsxs(Fragment, {
+      children: [contentCmp, jsx(SinglePageNavigation, { restoreScroll$ })],
```

Comment:
why another component? i am not sure i like the extra layer of indirection

Reply:
Sure, I will inline that into the RouterOutlet component

Reply:
done